src/backend/metal/src/device.rs

```diff
@@ -1769,7 +1820,20 @@ impl hal::device::Device<Backend> for Device {
             .map_err(|err| {
                 error!("PSO creation failed: {}", err);
                 pso::CreationError::Other
-            })
+            })?;
+
+        // We need to add the pipline descriptor to the binary archive after creating the
+        // pipeline, otherwise `new_render_pipeline_state_with_fail_on_binary_archive_miss`
+        // succeeds when it shouldn't.
+        if let Some(binary_archive) = cache.and_then(|cache| cache.binary_archive.as_ref()) {
```

Comment:
looks like this `cache.and_then(|cache| cache.binary_archive.as_ref())` could use a helper method